src/components/checkbox-group/checkbox-group.component.ts

```diff
@@ -0,0 +1,345 @@
+ */
+export default class SlCheckboxGroup extends ShoelaceElement implements ShoelaceFormControl {
+  static styles: CSSResultGroup = [componentStyles, formControlStyles, styles];
+
+  protected readonly formControlController = new FormControlController(this);
+  private readonly hasSlotController = new HasSlotController(this, 'help-text', 'label');
+  private customValidityMessage = '';
+  private validationTimeout: number;
+
+  @query('slot:not([name])') defaultSlot: HTMLSlotElement;
+  @query('.checkbox-group__validation-input') validationInput: HTMLInputElement;
+
+  @state() private errorMessage = '';
+
+  /**
+   * The checkbox group's label. Required for proper accessibility. If you need to display HTML, use the `label` slot instead. */
+  @property() label = '';
+
+  /** Text that appears in a tooltip next to the label. If you need to display HTML in the tooltip, use the `label-tooltip` slot instead. */
+  @property({ attribute: 'label-tooltip' }) labelTooltip = '';
+
+  /** The checkbox groups's help text. If you need to display HTML, use the `help-text` slot instead. */
+  @property({ attribute: 'help-text' }) helpText = '';
+
+  /** The name of the checkbox group, submitted as a name/value pair with form data. */
+  @property() name = '';
+
+  /**
+   * The current value of the checkbox group, submitted as a name/value pair with form data.
+   */
+  @property({ type: Array }) value: string[] = [];
+
+  /** The radio group's size. This size will be applied to all child radios and radio buttons. */
+  @property({ reflect: true }) size: 'small' | 'medium' | 'large' = 'medium';
+
+  /** The radio group's orientation. Changes the group's layout from the default (vertical) to horizontal. */
+  @property({ type: Boolean, reflect: true }) horizontal = false;
+
+  /** The radio group's style. Changes the group's style from the default (plain) style to the 'contained' style. This style will be applied to all child radios (but not child radio buttons, which do not have the 'contained' style as an option). */
+  @property({ type: Boolean, reflect: true }) contained = false;
+
+  /**
+   * By default, form controls are associated with the nearest containing `<form>` element. This attribute allows you
+   * to place the form control outside of a form and associate it with the form that has this `id`. The form must be in
+   * the same document or shadow root for this to work.
+   */
+  @property({ reflect: true }) form = '';
+
+  /** Ensures a child radio is checked before allowing the containing form to submit. */
+  @property({ type: Boolean, reflect: true }) required = false;
+
+  private initializeValueFromCheckboxes() {
+    const checkboxes = this.getAllCheckboxes();
+    this.value = checkboxes.map(checkbox => `${checkbox.value}: ${checkbox.checked}`);
+  }
+
+  /** Gets the validity state object */
+  get validity() {
+    if (this.value === null || this.value === undefined) {
+      this.initializeValueFromCheckboxes();
+    }
```

Comment:
Just my unfamiliarity here -- do we need this still despite it being called from `connectedCallback`?

Reply:
I added this because while all the tests were passing, I was getting this TypeError (see SS), and adding this `if` statement made it go away! I tried to adjust the tests a bunch to figure out why this is happening, but couldn't figure it out. If it's okay that I have this TypeError as long as the tests themselves are passing, then I can just delete this from the main component file, as everything is working as expected without this being repeated in the validity methods.
<img width="1130" alt="Screenshot 2024-05-06 at 16 05 44" src="https://github.com/teamshares/shoelace/assets/203228/f011bc87-3e77-4945-b60f-05022d2907b1">